### DISCUSSION TITLE
`Visibility` inheritance for ui entities

### DISCUSSION THREAD

**Discussion by nicopap** [bevyengine/bevy#3723] [Ideas]
In the current bevy ui crate. Would it make sense for the `Visibility` component to not only affect the entity it is bound to, but also all the children of that entity?

Typically, you want to hide/show complete menus when navigating between them, while having all accessible ones at least loaded. Currently, there are two ways to do that:
1. Have each menu physically separated and changing the camera position when switching menus (note: this is how _The Binding of Isaac: Rebirth_ does it)
2. Store somewhere (such as in a `Res`) the list of all entities in a given menu, and mutate _en masse_ the `Visibility.is_visible` for each entity in it

Having the `Visibility` component inheritable would make it only the encompassing menu node would need to be modified.

I guess I convinced myself option (1) is already more than enough. Ideally this could be combined with culling ui elements outside of the camera view, and there would be no reason to not do it.

Maybe other people may disagree with my conclusion. Any thought?

**Comment by Davier** [comment]
There is a third option that might fit better for this use case: setting the style's `display` to `Display::None`, which works recursively.

**Comment by cart** [comment]
We definitely want Visibility to "cascade down" hierarchy in Bevy (even for non ui things). This will be very important when nested scenes become a thing. Developers should be able to hide a  "player scene" by hiding the scene's root entity.
